mount/mount.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2018-2022 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//go:build !plan9
+// +build !plan9
+
+package mount
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/hashicorp/go-multierror"
+	"golang.org/x/sys/unix"
+)
+
+// This mounter type may be useful should we need more tests: we can call mount with a mock
+// mounter.
+type mounter func(source string, target string, fstype string, flags uintptr, data string) error
+
+// Mount takes a full fstab as a string and does whatever mounts are needed.
+// It ignores comment lines, and lines with less than 6 fields. In principal,
+// Mount should be able to do a full remount with the contents of /proc/mounts.
+func Mount(fstab string) error {
+	return mount(unix.Mount, fstab)
+}
+
+func mount(m mounter, fstab string) error {
+	var lineno int
+	s := bufio.NewScanner(strings.NewReader(fstab))
+	var err error
+	for s.Scan() {
+		lineno++
+		l := s.Text()
+		if strings.HasPrefix(l, "#") {
+			continue
+		}
+		f := strings.Fields(l)
+		// fstab is historical, pretty free format.
+		// Users may have dropped a random fstab in and we need
+		// to be forgiving.
+		// The last two fields no longer have any meaning or use.
```

Comment:
but we require them, even if they have no meaning for us.  it's not particularly 'forgiving' as the comment suggests.  i'm fine with whatever - just want to make sure we want to reject fstab lines that don't have the trailing 0 0.

Reply:
we do reject them. We can fix this comment as needed later I guess.